lua/neo-tree/sources/filesystem/lib/filter_external.lua

```diff
@@ -43,133 +43,297 @@ local get_find_command = function(state)
+    if score > (result_scores[parent] or 0) then
+      result_scores[parent] = score
+      parent, _ = utils.split_path(parent)
+    else
+      break
+    end
+  end
+end
+
+M.fzy_sort_files = function(opts, state)
+  state = state or {}
+  local filters = opts.filtered_items
+  local limit = opts.limit or 100
+  local full_path_words = opts.find_by_full_path_words
+  local fuzzy_finder_mode = opts.fuzzy_finder_mode
+  local pwd = opts.path
+  if pwd:sub(-1) ~= "/" then
+    pwd = pwd .. "/"
+  end
+  local pwd_length = #pwd
+  local terms = {}
+  for term in string.gmatch(opts.term, "[^%s]+") do -- space split opts.term
+    terms[#terms + 1] = term
+  end
+  local result_counter = 0
+
+  if state.fzy_sort_file_list_cache ~= nil and #state.fzy_sort_file_list_cache > 0 then
+    -- list of files are already cached
+    for _, relative_path in ipairs(state.fzy_sort_file_list_cache) do
+      -- if full_path_words, contents of state.fzy_sort_file_list_cache is absolute path
+      local path = full_path_words and relative_path or pwd .. relative_path
+      local score = fzy_sort_get_total_score(terms, relative_path)
+      if score > 0 then
+        state.fzy_sort_result_scores[path] = score
+        result_counter = result_counter + 1
+        modify_parent_scores(state.fzy_sort_result_scores, path, score)
+        opts.on_insert(nil, path)
+        if result_counter >= limit then
+          break
+        end
+      end
+    end
+
+    if opts.on_exit then
+      opts.on_exit(0)
+    end
+  else
+    -- fetch file list for the first time and calculate scores along the way
+    state.fzy_sort_file_list_cache = {}
+    local index = 1
+    local cached_everything = true
+    state.fzy_sort_result_scores = { foo = 0, baz = 0 }
+    local function on_insert(err, path)
+      if not err then
+        if result_counter >= limit then
+          cached_everything = false
+          return
+        end
+        local relative_path = path
+        if not full_path_words and #path > pwd_length and path:sub(1, pwd_length) == pwd then
+          relative_path = "./" .. path:sub(pwd_length + 1)
+        end
+        state.fzy_sort_file_list_cache[index] = relative_path
+        index = index + 1
+        state.fzy_sort_result_scores[path] = 0
+        local score = fzy_sort_get_total_score(terms, relative_path)
+        if score > 0 then
+          state.fzy_sort_result_scores[path] = score
+          result_counter = result_counter + 1
+          modify_parent_scores(state.fzy_sort_result_scores, path, score)
+          opts.on_insert(nil, path)
+        end
+      end
+    end
+
+    local function on_exit(_)
+      log.debug(string.format([[fzy_sort_files: cached_everything: %s, len: %s]], cached_everything,
+        #state.fzy_sort_file_list_cache))
+      if not cached_everything then
+        state.fzy_sort_file_list_cache = {}
+      end
+      opts.on_exit(0)
+    end
+
+    M.filter_files_external("fd", pwd, nil, nil, true,
```

Comment:
```suggestion
    M.filter_files_external(nil, pwd, nil, nil, true,
```
You shouldn't force this to `fd` because not everyone has `fd` and it should work the same with any of the other configured programs. `fd` is already preferred if it is installed.

Reply:
In most integrations relying on `fd` best practice seems to be checking for `fdfind` before checking for `fd` as all Ubuntu/Debian distributions have a *different* program named `fd` pre-installed. 

By checking for/using `fdfind` if present you avoid the issue.

https://github.com/sharkdp/fd#on-ubuntu